test/release/examples/primers/chplvis/chplvis.html

```diff
@@ -0,0 +1,312 @@
+<em>chplvis</em> is a tool to help the Chapel program visualize their
+Chapel program's tasks and communication between locales.  Using the
+standard module <em>VisualDebug</em>, the programmer controls what part
+of their program generates information for chplvis.  During the run of
+a program using the <em>VisualDebug</em> module, data files are
+created that are used as input for <em>chplvis</em>.  This document
+will help you understand the <em>VisualDebug</em> module and the
+<em>chplvis</em> tool.
+</p>
+
+<h3>Setup</h3>
+<p>
+<em>chplvis</em> is built by giving the command <em>"make chplvis"</em>
+at the top level of the chapel tree.  This also builds
+the GUI tool, <em>FLTK</em>, required to build and
+run <em>chplvis</em>.  (Note: Some versions of Linux may require the
+standard package <em>libx11-dev</em> to be installed before <em>FLTK</em> will
+compile properly.)  To get the most out of this primer, you should
+compile and run the example programs and examine
+the <em>VisualDebug</em> results with <em>chplvis</em>.  The graphics
+in this primer were produced on a system using the <em>fifo</em> package
+instead of <em>qthreads</em> for the tasking layer.   If you use
+<em>qthreads</em>, your task count may differ from the examples.
+</p>
+
+<h3>Chapel Source Code</h3>
+<p>
+To use <em>chplvis</em>, the programmer adds code to their program.  In many
+cases, the programmer may want to investigate only part of the program.  This
+is accomplished by having functions <em>startVdebug(name)</em> and
+<em>stopVdebug()</em> to control where to start and stop the instrumentation
+of their program.  Compilation and execution of these programs remain the
+same.  When the <em>startVdebug(name)</em> is executed, a collection
+of files, one per locale, are created in a directory with the <em>name</em> given
+in <em>startVdebug(name)</em>.
+</p>
+
+<h4>Example 1</h4>
+<p>
+Consider the chapel program: (prog1.chpl)
+<pre style="margin:20px">
+//  Example 1 using visual debug
+
+use VisualDebug;
+
+startVdebug("E1");
+
+coforall loc in Locales do
+   on loc do writeln("Hello from locale " + here.id + ".");
+
+stopVdebug();
+</pre>
+</p>
+
+<p>
+Compiling the program and running it with the options "-nl 6" will then
+produce a directory called <em>E1</em> containing 6 data files, one
+for each of the locales and named <em>E1-n</em> where <em>n</em> is
+replaced with the locale number, a number from 0 to 5.  Once this
+directory is created, one can run chplvis as <em>chplvis E1</em> or
+simply <em>chplvis</em> and then opening the file <em>E1/E1-0</em>
+from the <em>file/open</em> menu.  The resulting display is:
+</p>
+<img src="E1.png" style="margin:10px; border: 2px solid black;">
+<p>
+(Note: This image is from an X11 display.  On OS-X, the menu bar
+will be on the normal menu bar at the top of the screen and will
+not show in the main window.)
+</p>
+<h4>chplvis Elements</h4>
+<ul>
+<li><em>Locale</em> is represented by a box.  The initial display
```

Comment:
The English teacher in me would like an article before Locale and Data menu.  Communication links, being plural, is fine :)